commit 60ba38cf378d6980b345aa7d23bd1c6fa00d0bcd (refs/changes/94/3694/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-03-20T15:35:44-07:00 (1 year, 7 months ago)
    
    TRITON-178 Image creation support for bhyve VM
    Reviewed by: Trent Mick <trentm@gmail.com>

diff --git a/CHANGES.md b/CHANGES.md
index e93035e..20db701 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,10 @@
 # IMGAPI changelog
 
+## 4.3.0
+
+- TRITON-178 Add support for image creation with bhyve brand (requires platform
+  support)
+
 ## 4.2.2
 
 - TRITON-222 Use node-triton-metrics for IMGAPI metrics collection
diff --git a/docs/index.md b/docs/index.md
index 9f6b3d3..4ae74e6 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -1254,14 +1254,15 @@ from an existing VM:
 
 ### Inherited Fields
 
-The following is the list of fields that the new Image will inherit from the source
-Image of the VM in question and therefore cannot be specified:
+The following is the list of fields that the new image will inherit either
+(a) from the origin image (i.e. the image used to create this VM), or
+(b) from properites of the VM itself.
 
 | Field                                                    | Type    | Notes                                                                                                                                                                                           |
 | -------------------------------------------------------- | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | [type](#manifest-type)                                   | String  | The image type. One of "zone-dataset" for a ZFS dataset used to create a new SmartOS zone, "lx-dataset" for a Lx-brand image, "zvol" for a virtual machine image or "other" for image types that serve any other specific purpose. |
 | [os](#manifest-os)                                       | String  | The OS family this image provides. One of "smartos", "windows", and "linux".                                                                                                                    |
-| [requirements](#manifest-requirements)                   | Object  | A set of named requirements for provisioning a VM with this image. See [the requirements docs](#manifest-requirements) above for supported fields.                                              |
+| [requirements](#manifest-requirements)                   | Object  | A set of named requirements for provisioning a VM with this image. `requirements.min_platform` is set to the VM server's platform version for SmartOS VMs (where `vm.brand` is either "joyent" or "joyent-minimal"). `requirements.brand` is set to the VM's `brand` value for "lx", "kvm" and "bhyve" VMs. See [the requirements section](#manifest-requirements) above for details. |
 | [users](#manifest-users)                                 | Array   | A list of users for which passwords should be generated for provisioning. This may only make sense for some images. Example: `[{"name": "root"}, {"name": "admin"}]`                            |
 | [billing_tags](#manifest-billing-tags)                   | Array   | A list of tags that can be used by operators for additional billing processing.                                                                                                                 |
 | [traits](#manifest-traits)                               | Object  | An object that defines a collection of properties that is used by other APIs to evaluate where should customer VMs be placed.                                                                   |
diff --git a/lib/images.js b/lib/images.js
index 6905068..4ded88d 100644
--- a/lib/images.js
+++ b/lib/images.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -1880,7 +1880,7 @@ function apiCreateImageFromVm(req, res, callback) {
 
         function getPrepareImageScript(next) {
             var protoImageUuid;
-            if (vm.brand === 'kvm') {
+            if (['bhyve', 'kvm'].indexOf(vm.brand) !== -1) {
                 if (vm.disks) {
                     for (var i = 0; i < vm.disks.length; i++) {
                         if (vm.disks[i].image_uuid) {
diff --git a/package.json b/package.json
index 09101f7..ebd267e 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgapi",
   "description": "Image API to manage images for SDC 7",
-  "version": "4.2.2",
+  "version": "4.3.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
